cni/cmd/istio-cni/redirect.go

```diff
@@ -246,12 +248,19 @@ func NewRedirect(annotations map[string]string) (*Redirect, error) {
 	}
 	redir.excludeInboundPorts += "15020,15021,15090"
 	redir.excludeInboundPorts = strings.Join(dedupPorts(splitPorts(redir.excludeInboundPorts)), ",")
-	isFound, redir.kubevirtInterfaces, valErr = getAnnotationOrDefault("kubevirtInterfaces", annotations)
+	isFound, redir.kubevirtInterfaces, valErr = getAnnotationOrDefault("kubevirtInterfaces", pi.Annotations)
 	if valErr != nil {
 		log.Errorf("Annotation value error for value %s; annotationFound = %t: %v",
 			"kubevirtInterfaces", isFound, valErr)
 		return nil, valErr
 	}
+	if v, found := pi.ProxyConfig.ProxyMetadata[options.DnsCaptureByAgent.Name]; found {
```

Comment:
ProxyConfig may be more generally useful, but it seems we only use it for ProxyMetadata. Technically the agent doesn't read ProxyMetadata, it reads the env, which can be set by other means. Should we read the env possibly?

If we want the proxy config struct for other purposes we could "backfill" the env from the pod spec into ProxyConfig.ProxyMetadata I guess.

This should work in 95% of cases as is, just thinking about less common cases

Reply:
> ProxyConfig may be more generally useful, but it seems we only use it for ProxyMetadata.

Yeah I am thinking about reading include/exclude ip/port from proxyConfig, and avoid adding those annotations in sidecar injection templates.

>  Technically the agent doesn't read ProxyMetadata, it reads the env, which can be set by other means. Should we read the env possibly?

Yeah I am a bit hesitant to read from individual env variable, and I guess to make DNS capture beta, we will need some better first class API? But anyway I guess reading from env var is the most correct way to do it right now.

So for this change, I will keep ProxyConfig change for other usages, but read DNS capture information from env, wdyt?

Reply:
>  I am thinking about reading include/exclude ip/port from proxyConfig

Do we have those currently there? or in the future?

SGTM

Reply:
> Do we have those currently there? or in the future?

Ohh, they are just values.. Somehow I thought they were proxyConfig. Yeah, I guess one day in the future then :). Then I guess one thing we can clean up from sidecar template is interceptionMode, which is available at proxyConfig. Anyway, that's work for later.